lib/rust/ensogl/core/src/display/shape/primitive/style_watch.rs

```diff
@@ -84,36 +84,53 @@ impl StyleWatchFrp {
         sampler
     }
 
-    /// Queries style sheet value for a number.  Returns 0.0 if not found.
+    /// Queries style sheet value for a number. Emits a warning and returns 0.0 if not found.
     pub fn get_number(&self, path: impl Into<Path>) -> frp::Sampler<f32> {
         let network = &self.network;
+        let path = path.into();
+        let warning = format!("Tried to access undefined number from theme: {}", &path);
         let (source, current) = self.get_internal(path);
         frp::extend! { network
-            value   <- source.map(|t| t.number().unwrap_or(0.0));
+            value <- source.map(move |t| t.number().unwrap_or_else(|| {
+                WARNING!(warning);
+                0.0
+            }));
             sampler <- value.sampler();
         }
         source.emit(current);
         sampler
     }
 
-    /// Queries style sheet color, if not found fallbacks to [`FALLBACK_COLOR`].
+    /// Queries style sheet color, if not found fallbacks to [`FALLBACK_COLOR`] and emits a warning.
     pub fn get_color<T: Into<Path>>(&self, path: T) -> frp::Sampler<color::Rgba> {
         let network = &self.network;
+        let path = path.into();
+        let warning = format!("Tried to access undefined color from theme: {}", &path);
         let (source, current) = self.get_internal(path);
         frp::extend! { network
-            value   <- source.map(|t| t.color().unwrap_or(FALLBACK_COLOR));
+            value <- source.map(move |t| t.color().unwrap_or_else(|| {
+                WARNING!(warning);
```

Comment:
use tracing logger, not `WARNING`. See docs of `WARNING` - these screaming names are meant to be used as debug utils only.